searx/results.py

```diff
@@ -145,9 +145,9 @@ class ResultContainer:
     """docstring for ResultContainer"""
 
     __slots__ = '_merged_results', 'infoboxes', 'suggestions', 'answers', 'corrections', '_number_of_results',\
-                '_ordered', 'paging', 'unresponsive_engines', 'timings', 'redirect_url', 'engine_data'
+                '_closed', 'paging', 'unresponsive_engines', 'timings', 'redirect_url', 'engine_data', 'on_result'
 
-    def __init__(self):
+    def __init__(self, on_result=None):
```

Comment:
Sorry, I do not understand when is the argument on_result not None?  See my next question, it might be related

Reply:
`git grep "ResultContainer"`: 
* the checker creates ResultContainer without usage of the plugins
* `tests/unit/test_results.py` should be updated.

Reply:
> git grep "ResultContainer

That is what I have done, but I find only constructors without arguments `ResultContainer()`

Reply:
True...

https://github.com/dalf/searxng/blob/2b96ace61118d0a94eb74f6d078728a0c73276de/searx/search/__init__.py#L182 : 
* `self.result_container.on_result = self._on_result`

The parameter `on_result` is not given because the `Search` constructor creates the `ResultContainer`.

In the current design the `on_result` parameter can be removed.